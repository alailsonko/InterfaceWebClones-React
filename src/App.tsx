import React from 'react';

export interface Props {
  name?: string;
}
function App(props: Props) {
  const { name } = props;
  if (name) {
    return (
      <h1>
        Hello,
        {' '}
        {name}
        !
      </h1>
    );
  }
  return <span>Hey, stranger</span>;
}

export default App;
