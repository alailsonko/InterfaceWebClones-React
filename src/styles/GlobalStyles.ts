import { createGlobalStyle } from 'styled-components';

const GlobalStyles = createGlobalStyle`

:root {
  --color-black-lighter: #5B575C;
  --color-purple-lighter: #D992E1;
  --color-gray-lighter: #DAD1DB;
  --color-purple-darker: #593C5C;
  --color-gray-darker: #A7A0A8;
}

html, body {
  background-color: var(--color-gray-lighter);
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

h1, h2, h3, h4, button {
  font-family: 'Roboto', sans-serif;
  font-weight: 700;
}

p { 
  font-family: 'Roboto', serif;
}

input {
  outline: none;
}

`;

export default GlobalStyles;
