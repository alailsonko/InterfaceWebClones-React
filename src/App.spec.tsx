import { shallow } from 'enzyme';
import React from 'react';
import Routes from 'routes';
import GlobalStyles from 'styles/GlobalStyles';
import App from './App';

const wrapper = shallow(<App />);

describe('App component', () => {
  test('should render GlobalStyles and Routes', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('should match component GlobalStyles', () => {
    expect(wrapper.find(<GlobalStyles />)).toMatchSnapshot();
  });
  test('should match component Routes', () => {
    expect(wrapper.find(<Routes />)).toMatchSnapshot();
  });
});
