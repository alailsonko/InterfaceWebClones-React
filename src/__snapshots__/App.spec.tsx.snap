// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should match component GlobalStyles 1`] = `null`;

exports[`App component should match component Routes 1`] = `null`;

exports[`App component should render GlobalStyles and Routes 1`] = `
<Fragment>
  <Routes />
  <Memo(l) />
</Fragment>
`;
